File: netapp-manageability-sdk-9.3/src/sample/DataFabric_Manager/API_Sample_Code/basic/Perl_Bindings/perl/bin/perl_bindings/sample/DataFabric_Manager/API_Sample_Code/basic/hello_dfm.pl
#===============================================================#
#                                                               #
# $ID$                                                          #
#                                                               #
# hello_dfm.pl                                                  #
#                                                               #
# Copyright (c) 2013 NetApp, Inc. All rights reserved.          #
# Specifications subject to change without notice.              #
#                                                               #
# This program will print the version number of the OCUM Server #
#                                                               #
# This Sample code is supported from OnCommand Unified Manager  #
# 6.0 onwards.                                                  #
#===============================================================#
require 5.6.1;

use lib '../../../../../../lib/perl/NetApp';
use NaServer;

# Variables declaration
my $args = $#ARGV + 1;
my $ocumserver = shift;
my $ocumuser = shift;
my $ocumpw	= shift;

# check for valid number of parameters
if ($args != 3) {
	print_usage();
}

my $s = NaServer->new ($ocumserver, 1, 0);
$s->set_style(LOGIN);
$s->set_admin_user($ocumuser, $ocumpw);
$s->set_server_type(OCUM);

eval {
    our $output = $s->system_about();
};

if($@) {
    print "Failed : $@";
} 
else {
    our $version = $output->{'version'};
    print "Hello world!  OCUM Server version is: $version\n";
}
################################################################################
sub print_usage()
{
	print "Usage: hello_dfm.pl <ocumserver> <ocumuser> <ocumpassword> \n";
	print "<ocumserver> -- Name/IP Address of the OCUM server \n";
	print "<ocumuser> -- OCUM server User name\n";
	print "<ocumpassword> -- OCUM server Password\n";
	exit -1;
}


#=========================== POD ============================#

=head1 NAME

  hello_dfm.pl - Gets OCUM server version


=head1 SYNOPSIS

  hello_dfm.pl  <ocumserver> <ocumuser> <ocumpassword>

=head1 ARGUMENTS

  <ocumserver>
   OCUM server name.

  <ocumuser>
  OCUM server username.

  <ocumpassword>
  OCUM server password.

  
=head1 SEE ALSO

  NaElement.pm, NaServer.pm, OCUMAPI.pm

=head1 COPYRIGHT

 Copyright (c) 2013 NetApp, Inc. All rights reserved.
 Specifications subject to change without notice.

=cut

File: netapp-manageability-sdk-9.3/src/sample/DataFabric_Manager/API_Sample_Code/basic/Python/hello_dfm.py
#===============================================================#
#                                                               #
# $ID$                                                          #
#                                                               #
# hello_dfm.py                                                  #
#                                                               #
# Copyright (c) 2011 NetApp, Inc. All rights reserved.          #
# Specifications subject to change without notice.              #
#                                                               #
# This program will print the version number of the DFM Server  #
#                                                               #
# This Sample code is supported from DataFabric Manager 3.6R2   #
# onwards.                                                      #
# However few of the functionalities of the sample code may     #
# work on older versions of DataFabric Manager.                 #
#===============================================================#

import sys
sys.path.append("../../../../../../lib/python/NetApp")
from NaServer import *


def print_usage():
    print ("Usage: hello_dfm.py <dfmserver> <dfmuser> <dfmpassword> \n")
    print ("<dfmserver> -- Name/IP Address of the DFM server \n")
    print ("<dfmuser> -- DFM server User name\n")
    print ("<dfmpassword> -- DFM server Password\n")
    sys.exit (1)

args = len(sys.argv)-1

if(args != 3):
    print_usage()

dfmserver = sys.argv[1]
dfmuser = sys.argv[2]
dfmpw = sys.argv[3]
s = NaServer(dfmserver, 1, 0)
s.set_style('LOGIN')
s.set_transport_type('HTTP')
s.set_server_type('DFM')
s.set_port(8088)
s.set_admin_user(dfmuser, dfmpw)
output = s.invoke("dfm-about")

if(output.results_errno() != 0):
    r = output.results_reason()
    print("Failed " + str(r) + "\n")

else :
    r = output.child_get_string("version")
    print("Hello World ! DFM Server version is: " + r + "\n")


